llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -64,16 +64,19 @@ class Value;
 //
 // If asked to represent a pathologically large value, this will degrade to
 // std::nullopt.
+// Store Scalable information in bit 62 of Value. Scalable information is
+// required to do Alias Analysis on Scalable quantities
 class LocationSize {
   enum : uint64_t {
     BeforeOrAfterPointer = ~uint64_t(0),
-    AfterPointer = BeforeOrAfterPointer - 1,
+    ScalableBit = uint64_t(1) << 62,
+    AfterPointer = (BeforeOrAfterPointer - 1) & ~ScalableBit,
```

Comment:
Why does `AfterPointer` have the ScalableBit cleared but none of the other sentinel values do?

Reply:
https://github.com/llvm/llvm-project/pull/69716#discussion_r1368450981

some tests seem to fail if I don't exclude the `ScalableBit` from `AfterPointer`